test/integration_test.exs

```diff
@@ -156,82 +193,130 @@ defmodule IntegrationTest do
     assert result.columns == ["en_description"]
     assert result.num_rows == 1
 
-    assert result.rows == [["fruit of the apple tree"]]
+    rows = result.rows |> Enum.to_list()
+    assert rows == [["fruit of the apple tree"]]
   end
 
   test "encodes and decodes types received from Google BigQuery's response", %{
     test: goth
   } do
-    project_id = System.fetch_env!("PROJECT_ID")
-
     credentials =
       System.get_env("GOOGLE_APPLICATION_CREDENTIALS", "credentials.json")
       |> File.read!()
       |> Jason.decode!()
 
+    project_id = System.get_env("PROJECT_ID", credentials["project_id"])
+
     source = {:service_account, credentials, []}
     start_supervised!({Goth, name: goth, source: source, http_client: &Req.request/1})
 
     req = Req.new() |> ReqBigQuery.attach(project_id: project_id, goth: goth)
 
     value = Decimal.new("1.1")
-    assert Req.post!(req, bigquery: {"SELECT ?", [value]}).body.rows == [[value]]
+
+    assert Req.post!(req, bigquery: {"SELECT ?", [value]}).body.rows |> Enum.to_list() == [
+             [value]
+           ]
```

Comment:
For these repetitive integration tests, what if we create a simple function to help on how it would work?

```elixir
defp run_decoding_query(req, input) do
  result = Req.post!(req, bigquery: {"SELECT ?", [input]}).body
  Enum.to_list(result.rows)
end
```

Making it possible:

```suggestion
    assert run_decoding_query(req, value) == [[value]]
```

WDYT?